CalendarApp/Logic Handlers/SyncConflictHandler.swift

```diff
@@ -7,13 +7,204 @@
 
 import Foundation
 
+@objc
+public protocol SyncChangesDelegate {
+    func deletedEventOnRemote(eventID: UUID)
+    func updatedEventOnRemote(event: Event)
+    func createdEventOnRemote(eventID: UUID)
+    
+    func syncNewEventToParse(event: Event, remoteChange: RemoteChange)
+    func syncDeleteToParse(event: String, remoteChange: RemoteChange)
+    func syncUpdateToParse(event: Event)
+    func syncRemoteChangeToParse(remoteChange: RemoteChange)
+}
+
 @objcMembers
 class SyncConflictHandler : NSObject {
-    private var keptChanges = [LocalChange]()
+    weak var delegate: SyncChangesDelegate?
+    private var revisionHistories: [[Revision]]
+    private let localChangeHandler = LocalChangeHandler()
+    private let parseChangeHandler = ParseChangeHandler()
+    private let coreDataEventHandler = CoreDataEventHandler()
+    
+    init(histories: [[RemoteChange]]) {
+        revisionHistories = histories
+    }
+    
+    func syncChanges() {
+        addLocalChanges()
+        checkChanges()
+        syncLocalCreatesDeletes()
+        localChangeHandler.deleteAllLocalChanges()
+    }
+    
+    private func syncLocalCreatesDeletes() {
+        let changes = localChangeHandler.fetchAllLocalChanges()
+        for change in changes {
+            if change.changeType == .Create {
+                guard let eventID = change.eventID,
+                      let event = coreDataEventHandler.queryEvent(from: eventID) else {
+                    break
+                }
+                delegate?.syncNewEventToParse(event: event, remoteChange: getRemoteChange(localChange: change))
+            } else if (change.changeType == .Delete) {
+                guard let eventID = change.eventID else {
+                    break
+                }
+                delegate?.syncDeleteToParse(event: eventID.uuidString,
+                                            remoteChange: getRemoteChange(localChange: change))
+            }
+        }
+    }
+    
+    private func addLocalChanges() {
+        if (revisionHistories.count == 0) {
+            return
+        }
+        for historyIndex in 0...revisionHistories.count - 1 {
+            guard let eventID = revisionHistories[historyIndex][0].eventID else {
```

Comment:
Use forEach and you won't have to index this way. This is scary to see in swift :)